core/shared/platform/common/posix/platform_api_posix.cmake

```diff
@@ -9,6 +9,11 @@ if (NOT WAMR_BUILD_LIBC_WASI EQUAL 1)
     list(REMOVE_ITEM source_all
         ${PLATFORM_COMMON_POSIX_DIR}/posix_file.c
         ${PLATFORM_COMMON_POSIX_DIR}/posix_clock.c
+    )
+endif()
+
+if ((NOT WAMR_BUILD_LIBC_WASI EQUAL 1) AND (NOT WAMR_BUILD_DEBUG_INTERP EQUAL 1))

```

Comment:
This causes the condition for the `else()` branch below differs from the original one, how about:
```cmake
if (NOT WAMR_BUILD_LIBC_WASI EQUAL 1)
    list(REMOVE_ITEM source_all
        ${PLATFORM_COMMON_POSIX_DIR}/posix_file.c
        ${PLATFORM_COMMON_POSIX_DIR}/posix_clock.c
    )
    if (NOT WAMR_BUILD_DEBUG_INTERP EQUAL 1)
        list(REMOVE_ITEM source_all
             ${PLATFORM_COMMON_POSIX_DIR}/posix_socket.c
        )
    endif()
else()
    ...
endif()
```